R/object_usage_linter.R

```diff
@@ -45,34 +45,49 @@ object_usage_linter <- function() {
       }
       res <- parse_check_usage(fun)
 
-      lapply(which(!is.na(res$message)),
-             function(row_num) {
-               row <- res[row_num, ]
-
-               if (row$name %in% declared_globals) {
-                 return()
-               }
-
-               org_line_num <- as.integer(row$line_number) + info$line1 - 1L
-
-               line <- source_file$content[as.integer(org_line_num)]
-
-               row$name <- re_substitutes(row$name, rex("<-"), "")
-
-               location <- re_matches(line,
-                                      rex(row$name),
-                                      locations = TRUE)
-
-               Lint(
-                 filename = source_file$filename,
-                 line_number = org_line_num,
-                 column_number = location$start,
-                 type = "warning",
-                 message = row$message,
-                 line = line,
-                 ranges = list(c(location$start, location$end))
-               )
-             })
+      lapply(
+        which(!is.na(res$message)),
+        function(row_num) {
+          row <- res[row_num, ]
+
+          if (row$name %in% declared_globals) {
+            return()
+          }
+
+          org_line_num <- as.integer(row$line1) + info$line1 - 1L
+          line <- source_file$content[as.integer(org_line_num)]
+
+          row$name <- re_substitutes(row$name, rex("<-"), "")
+
+          location <- re_matches(line, rex(boundary, row$name, boundary), locations = TRUE)
+
+          if (is.na(location$start) && nzchar(row$line2) && row$line2 != row$line1) {
```

Comment:
Could you add a quick comment here explaining this branch? At a minimum pointing to the issue #

Reply:
Done.